prql-compiler/src/sql/dialect.rs

```diff
@@ -109,7 +110,7 @@ pub(super) enum ColumnExclude {
     Except,
 }
 
-pub(super) trait DialectHandler {
+pub(super) trait DialectHandler: Any {
```

Comment:
Could I ask why this is needed? (You probably know more than me about the complexities of rust's type system...)

Reply:
the type_id thing is granted by std::any (https://doc.rust-lang.org/std/any/index.html). I learnt this trick from https://github.com/sqlparser-rs/sqlparser-rs/blob/0c0d088ec21d7f3883165d36c34300745816c5c7/src/dialect/mod.rs#L110-L113

Reply:
Great, thanks for the reference! TIL